base/promotion.jl

```diff
@@ -243,10 +245,22 @@ function sametype_error(input...)
           " failed to change any input types")
 end
 
-+(x::Number, y::Number) = +(promote(x,y)...)
-*(x::Number, y::Number) = *(promote(x,y)...)
--(x::Number, y::Number) = -(promote(x,y)...)
-/(x::Number, y::Number) = /(promote(x,y)...)
+function +(x::Number, y::Number)
+    xp, yp = promote(x, y)
```

Comment:
Yes, I really don't like having to expand these like this – it's very ugly and we've never had to do so before.